common/debug_interface/src/json_log.rs

```diff
@@ -0,0 +1,50 @@
+use lazy_static::lazy_static;
+use serde_json::{self, value as json};
+use std::{collections::VecDeque, sync::Mutex};
+
+pub struct JsonLogEntry {
+    pub name: &'static str,
+    pub json: json::Value,
+}
+
+const MAX_EVENTS_IN_QUEUE: usize = 1_000;
+
+/// Writes event to event stream
+/// Example:
+///   event!("committed", block="b");
+#[macro_export]
+macro_rules! event {
+    ($name:expr, $($json:tt)*) => {
+        $crate::json_log::send_json_log($crate::json_log::JsonLogEntry{
+            name: $name,
+            json: serde_json::json!({$($json)+}),
+        });
+    };
+}
+
+lazy_static! {
+    // This queue maintains last MAX_EVENTS_IN_QUEUE events
+    // This is very efficiently implemented with circular buffer with fixed capacity
+    static ref JSON_LOG_ENTRY_QUEUE: Mutex<VecDeque<JsonLogEntry>> = Mutex::new(
```

Comment:
[Future suggestion] We could consider having a map with a queue per event type, it could make this design more modular to allow different fine grained debug queries (e.g., for specific event types).